### PR TITLE
Backend: Add feature for challenge host to toggle the leaderboard sorting order

### PR DESCRIPTION
**Pull request by Sanji515**
**Issue:**

Challenge host unable to toggle the leaderboard sorting order.

**Fix:**

Added backend feature to toggle the leaderboard sorting order by challenge host

#### Changes proposed in this pull request:

- [x] Added new field `leaderboard_order_is_decreasing field` in `ChallengePhaseSplit` model.

- [x] Added this field for the admin and `ZipChallengePhaseSplitSerializer`.

- [x] Added a condition to toggle the sorting order a/c to this field in the leaderboard API.

- [x] Update the doc `configuration.md` for having `leaderboard_order_is_decreasing` field under `challenge_phases_split` of `challenge_config.yml` file

- [x] Updated the backend tests

### CODE REVIEW

File: docs/source/configuration.md [hunk 1]
```diff
@@ -114,6 +114,8 @@ Following fields are required (and can be customized) in the [`challenge_config.
 
   - **visibility**: It will set the visibility of the numbers corresponding to metrics for this `challenge_phase_split`. Select one of the following positive integers based on the visibility level you want:
 
+  - **leaderboard_order_is_descending**: True/False (a Boolean field that gives the flexibility to Challenge Hosts to toggle the leaderboard sorting order. Default is `True`)
```
Comment (RishabhJain2018) [inline docs/source/configuration.md]: `True/False (a Boolean field that gives the flexibility to Challenge Hosts to toggle the leaderboard sorting order. Default is `True`)` -> `True/False (a Boolean field that gives the flexibility to challenge host for changing the default leaderboard sorting order. Default is `True`)`

Comment (Sanji515) [reply docs/source/configuration.md]: @RishabhJain2018 done

File: apps/challenges/admin.py [hunk 2]
```diff
@@ -92,6 +92,7 @@ class ChallengePhaseSplitAdmin(ImportExportTimeStampedAdmin):
         "dataset_split",
         "leaderboard",
         "visibility",
+        "leaderboard_order_is_descending"
```
Comment (deshraj) [inline apps/challenges/admin.py]: Let's change the field name in the model to `is_leaderboard_order_descending`

File: docs/source/configuration.md [hunk 3]
```diff
@@ -114,6 +114,8 @@ Following fields are required (and can be customized) in the [`challenge_config.
 
   - **visibility**: It will set the visibility of the numbers corresponding to metrics for this `challenge_phase_split`. Select one of the following positive integers based on the visibility level you want:
 
+  - **leaderboard_order_is_descending**: True/False (a Boolean field that gives the flexibility to challenge host for changing the default leaderboard sorting order. Default is `True`)
```
Comment (deshraj) [inline docs/source/configuration.md]: Let's change this to:

>   - **is_leaderboard_order_descending**: True/False (a Boolean field that gives the flexibility to challenge host to change the default leaderboard sorting order. It is useful in cases where you have `error` as a metric and want to sort the leaderboard in increasing order of `error` value. Default is `True`)

**Comment by deshraj** [comment]
Overall looks good. Good job @Sanji515. 

Also, can you please do the same changes in the config of https://github.com/Cloud-CV/EvalAI-Starters and update the config.

File: apps/challenges/admin.py (hunk 2 above)
Comment (Sanji515) [reply apps/challenges/admin.py]: @deshraj done

File: docs/source/configuration.md (hunk 3 above)
Comment (Sanji515) [reply docs/source/configuration.md]: done

**Comment by Sanji515** [comment]
Thanks @deshraj for reviewing the PR. Made the changes here and in `EvalAI-starters` repo, please take a look

**Pull request merged**